rocksdb_admin/admin_handler.cpp

```diff
@@ -81,6 +89,43 @@ namespace {
 const int kMB = 1024 * 1024;
 const int kS3UtilRecheckSec = 5;
 
+const int64_t kMillisPerSec = 1000;
+const char kKafkaConsumerType[] = "rocksplicator_message_consumer";
+const char kKafkaWatcherName[] = "RocksplicatorMessageConsumer";
+const uint32_t kKafkaConsumerPoolSize = 1;
+
+// Global condition variables for kafka synchronization.
+struct SyncVariables
+{
+  SyncVariables(): flag(false) {} // init flag to false
+  std::atomic<bool> flag;
+  std::condition_variable cond_var;
+};
+std::unordered_map<std::string, std::shared_ptr<SyncVariables>> sync_var_map;
+
+// Global map to keep track of serverset file watchers
+std::unordered_map<std::string, std::shared_ptr<
+    ::kafka::KafkaBrokerFileWatcher>> serverset_path_to_file_watcher_map;
+
+int64_t GetMessageTimestampSecs(const RdKafka::Message& message) {
+  const auto ts = message.timestamp();
+  if (ts.type == RdKafka::MessageTimestamp::MSG_TIMESTAMP_CREATE_TIME) {
+    return ts.timestamp / kMillisPerSec;
+  }
+
+  // We only expect the timestamp to be create time.
+  return -1;
+}
+
+std::string ToUTC(const int64_t time_secs) {
```

Comment:
I am seeing that this is only used once when printing log, can we use std::put_time directly there? https://en.cppreference.com/w/cpp/io/manip/put_time

Reply:
I'm planning to keep these logs only while we debug the system. In addition, `put_time` would also require converting epoch timestamp to time_t. So will keep this for now.